build-system/eslint-rules/no-export-side-effect.js

```diff
@@ -19,22 +19,23 @@ module.exports = function(context) {
   return {
     ExportNamedDeclaration: function(node) {
       if (node.declaration) {
-        var declaration = node.declaration;
+        const {declaration} = node;
         if (declaration.type === 'VariableDeclaration') {
           declaration.declarations
-              .map(function(declarator) {
-                return declarator.init
-              }).filter(function(init) {
-                return init && /(?:Call|New)Expression/.test(init.type)
-                    // Special case creating a map object from a literal.
-                    && init.callee.name != 'map';
-              }).forEach(function(init) {
-                context.report(init, 'Cannot export side-effect');
+            .map(function(declarator) {
+              return declarator.init;
+            })
+            .filter(function(init) {
+              return init && /(?:Call|New)Expression/.test(init.type);
+            })
+            .forEach(function(init) {
+              context.report({
+                node: init,
+                message: 'Cannot export side-effect',
               });
+            });
         }
-      } else if (node.specifiers) {
-        context.report(node, 'Side-effect linting not implemented');
       }
     },
   };
-};
+};
```

Comment:
can you add new line and in the description what this PR allows us to do or not do.. some examples maybe and describe the motivation for this change?

Reply:
I missed that. We are syncing from amphtml, is that all?

Reply:
Yep, earlier today I made a PR to `amphtml`. It basically allows us to use the `export {obj as deprecatedObj}` pattern.

Reply:
Just updated the description :)